cmd/ndm_daemonset/probe/sysfsprobe.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2020 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+
+Find queue sysfs doc here.
+https://www.kernel.org/doc/Documentation/block/queue-sysfs.txt
+
+*/
+
+package probe
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	"github.com/openebs/node-disk-manager/blockdevice"
+	"github.com/openebs/node-disk-manager/cmd/ndm_daemonset/controller"
+	"github.com/openebs/node-disk-manager/pkg/util"
+	"k8s.io/klog"
+)
+
+const (
+	sysfsProbePriority = 2
+	sysfsConfigKey     = "sysfs-probe"
+	// sectorSize is the sector size as understood by the unix systems.
+	// It is kept as 512 bytes. all entries in /sys/class/block/sda/size
+	// are in 512 byte blocks
+	sectorSize int64 = 512
+)
+
+var (
+	sysfsProbeName  = "sysfs probe"
+	sysfsProbeState = defaultEnabled
+)
+
+var sysfsProbeRegister = func() {
+	ctrl := <-controller.ControllerBroadcastChannel
+	if ctrl == nil {
+		klog.Error("unable to configure", sysfsProbeName)
+		return
+	}
+	if ctrl.NDMConfig != nil {
+		for _, probeConfig := range ctrl.NDMConfig.ProbeConfigs {
+			if probeConfig.Key == sysfsConfigKey {
+				sysfsProbeName = probeConfig.Name
+				sysfsProbeState = util.CheckTruthy(probeConfig.State)
+				break
+			}
+		}
+	}
+
+	newRegistryProbe := &registerProbe{
+		priority:   sysfsProbePriority,
+		name:       sysfsProbeName,
+		state:      sysfsProbeState,
+		pi:         newSysFSProbe(),
+		controller: ctrl,
+	}
+	newRegistryProbe.register()
+}
+
+// sysfsProbe fills the logical sector size,
+// physical sector size, drive type(ssd or hdd) of the disk
+type sysfsProbe struct {
+}
+
+func newSysFSProbe() *sysfsProbe {
+	return &sysfsProbe{}
+}
+
+// It is part of probe interface. Hence, empty implementation.
+func (cp *sysfsProbe) Start() {}
+
+// FillBlockDeviceDetails updates the logical sector size,
+// physical sector size, drive type(ssd or hdd) of the disk
+// if those are not populated.
+func (cp *sysfsProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockDevice) {
+	sysPath := blockDevice.SysPath
+
+	if blockDevice.DeviceAttributes.LogicalBlockSize == 0 {
+		logicalBlockSize, err := readSysFSFileAsInt64(sysPath + "/queue/logical_block_size")
+		if err != nil {
+			klog.Warning("unable to read logical block size", err)
+		} else if logicalBlockSize != 0 {
+			blockDevice.DeviceAttributes.LogicalBlockSize = uint32(logicalBlockSize)
```

Comment:
should this be uint64 for capacity higher than 4T that we have seen in #150

Reply:
I confused this with Capacity. But would be good to know what is the max value that can be seen here. If it is like 512, 4k, 16k. we can leave it as uint32.

Reply:
I have seen disk with 16k physical block size and max 4k logical block size.